### PR TITLE
Add pagination

### PR DESCRIPTION
**Pull request by rkondratowicz**
### Change description ###
Adding pagination, as required by https://hmcts.github.io/restful-api-standards/#pagination

**Does this PR introduce a breaking change?** (check one with "x")
```
[x] Yes
[ ] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/draftstore/data/PaginationTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+        // then
+        assertThat(drafts).hasSize(limit);
+    }
+
+    @Test
+    public void should_return_elements_after_given_position() throws Exception {
+        // given
+        List<Integer> createdDraftIds = repeat(10, () -> insert());
+        Integer fifthElement = createdDraftIds.get(4);
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/draftstore/data/PaginationTest.java]: pun intended

File: src/main/java/uk/gov/hmcts/reform/draftstore/endpoint/v3/DraftController.java [hunk 2]
```diff
@@ -79,10 +80,12 @@ public Draft read(
     })
     public DraftList readAll(
         @RequestHeader(AUTHORIZATION) String authHeader,
-        @RequestHeader(SERVICE_HEADER) String serviceHeader
+        @RequestHeader(SERVICE_HEADER) String serviceHeader,
+        @RequestParam(name = "after", required = false) Integer after,
+        @RequestParam(name = "limit", required = false, defaultValue = "10") int limit
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/reform/draftstore/endpoint/v3/DraftController.java]: potentially a new version of the API needed for this?

File: src/functionalTest/java/uk/gov/hmcts/reform/draftstore/data/PaginationTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest
+@ActiveProfiles("test")
+public class PaginationTest {
+
+    @Autowired
+    private DraftStoreDAO repo;
+
+    @Test
+    public void should_limit_number_of_results() throws Exception {
```
Comment (KarolJastrzebski) [inline src/functionalTest/java/uk/gov/hmcts/reform/draftstore/data/PaginationTest.java]: Not sure if checked generic Exception is necessary. Looks like could be removed.

File: src/main/java/uk/gov/hmcts/reform/draftstore/endpoint/v3/DraftController.java (hunk 2 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/draftstore/endpoint/v3/DraftController.java]: we can change it to return everything if `limit` query param is not added

File: src/functionalTest/java/uk/gov/hmcts/reform/draftstore/data/PaginationTest.java (hunk 3 above)
Comment (rkondratowicz) [reply src/functionalTest/java/uk/gov/hmcts/reform/draftstore/data/PaginationTest.java]: That's just what Intellij generated by default (I'm not sure why though :)).
Most test methods in this project have such test method signatures.

Comment (timja-kainos) [reply src/functionalTest/java/uk/gov/hmcts/reform/draftstore/data/PaginationTest.java]: its just in the default intellij template for creating tests.
Probably fine

File: src/main/java/uk/gov/hmcts/reform/draftstore/data/DraftStoreDAO.java [hunk 4]
```diff
@@ -125,12 +125,19 @@ public void update(int id, UpdateDraft draft) {
...
     }
 
-    public List<Draft> readAll(String userId, String service) {
+    public List<Draft> readAll(String userId, String service, Integer after, int limit) {
         return jdbcTemplate.query(
-            "SELECT * FROM draft_document WHERE user_id = :userId AND service = :service",
+            "SELECT * FROM draft_document "
+                + "WHERE user_id = :userId "
+                + "AND service = :service "
+                + Optional.ofNullable(after).map(a -> "AND id > :after ").orElse("")
```
Comment (KarolJastrzebski) [inline src/main/java/uk/gov/hmcts/reform/draftstore/data/DraftStoreDAO.java]: Why pagination is based on primary key values instead of offset?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/draftstore/data/DraftStoreDAO.java]: check link from pull request description

**Pull request merged**